tensorflow/python/ops/array_ops.py

```diff
@@ -3628,7 +3645,114 @@ def _get_paddings_constant(paddings):
     return [_get_paddings_constant(x) for x in paddings]
   else:
     return paddings
+  
+def _wrap_pad(tensor, paddings, name=None):
+  """Helper to create wrap pad option with ops already built.
 
+  See `pad` for more info on how this function works.
+  """
+  # Use constant padding to create a tensor with final shape and original values.
+  padded = gen_array_ops.pad(input=tensor, paddings=paddings, name=name)
+  start_coords, end_coords = [
+    shape(padded) - paddings[:,1] - shape(tensor), 
+    shape(padded) - paddings[:,1]
+    ]
+  tensor_to_copy = tensor
+  strides = ones(rank(padded))
+  # Add wraps for each dimension.
+  for i in range(shape(paddings)[0]):
+    # The shape of `tensor_to_copy` can change in iterations. 
+    # Set the loop options to avoid an error in graph 
+    # mode.
+    directives.set_loop_options(
+                shape_invariants=[(tensor_to_copy, 
+                                   tensor_shape.TensorShape(None))]
+            )
+    lower_idx, upper_idx = paddings[i][0], paddings[i][1]
+    lower_begin = identity(start_coords)
+    upper_begin = identity(start_coords)
+    # Add the lower wrap.
+    while lower_idx > 0:
+      # Check for partial wrap.
+      if lower_idx < shape(tensor)[i]:
+        partial_slice_begin = zeros(rank(tensor_to_copy), dtype=dtypes.int32)
+        partial_slice_begin = gen_array_ops.tensor_scatter_add(
+          partial_slice_begin, 
+          reshape(i, [1,1]), 
+          (shape(tensor_to_copy)[i] - lower_idx)[..., None]
+          )
+        lower_size = fill(reshape(rank(tensor_to_copy), [-1]),-1)
+        length_to_update, value  = [
+          lower_idx, 
+          slice(tensor_to_copy, partial_slice_begin, lower_size)
+          ]
+      else:
+        length_to_update, value = [shape(tensor)[i], tensor_to_copy]
+      lower_begin = gen_array_ops.tensor_scatter_sub(
+        lower_begin, 
+        reshape(i, [1,1]), 
+        length_to_update[...,None]
+        )
+      end = lower_begin + shape(value)
+      # Pad tensor with one iteration.
+      padded = gen_array_ops.tensor_strided_slice_update(
+        input=padded, 
+        begin=gen_math_ops.cast(lower_begin, dtype=dtypes.int32), 
```

Comment:
Use `math_ops.cast` instead of  `gen_math_ops` - the `Cast` op doesn't have a "dtype" parameter, instead it has `SrcT` and `DstT` attributes.

Reply:
I think the parameters are the same, it's jsut not called dtype. You can see other instances of the same function being used here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/array_ops.py#L1147-L1161

Reply:
I've updated this to remove `dtype=`